### ISSUE TITLE
Cannot get final name for export 'BarChart4Icon' of __barrel_optimize__

### ISSUE THREAD

**Issue by omgismartinez** [vercel/next.js#58756]
### Link to the code that reproduces this issue

https://github.com/omgismartinez/eye/blob/dev/config/site.ts

### To Reproduce

1. Start the build of the application (next build)

### Current vs. Expected behavior

It is only an object where it is specified which is the icon that makes reference to the route, the `lucide-react` library is used. 
![image](https://github.com/vercel/next.js/assets/72507996/38ba7aa4-4a78-43d5-b7c0-c08761c54c11)

In development it works fine but when building it fails.
![image](https://github.com/vercel/next.js/assets/72507996/8c26a323-a6d0-4a48-abf2-53d6c270dba2)

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home  
Binaries:
  Node: 21.2.0
  npm: N/A
  Yarn: N/A
  pnpm: 8.10.5
Relevant Packages:
  next: 14.0.3
  eslint-config-next: 14.0.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure, SWC minifier (swcMinify: true), SWC transpilation

### Additional context

1. I tried with version 13.5.x and now with version 14.0.3 but it keeps crashing,
2. I tried to change the file extension from `.ts` to `.tsx` but the error persists. 
3. Change the different icon imports of lucide-react (BarChart4Icon, BarChart4)
4. The application is deployed in Vercel (error persists)

**Comment by fredereeks** [comment]
I found the solution (at least, for me). I had ```import {IoIosTrendingUp, IoIosCode, IoIosLaptop } from 'react-icons/io``` which works in development but fails on build. 

After much effort, I inspected the folder and file structures of the react-icons library, then, found all the icons were also exported in the index page (also in the index.esm but that isn't useful) and imported the font/icons from it (the index.js page). Short story long, I modified the above import line to ```import {IoIosTrendingUp, IoIosCode, IoIosLaptop } from 'react-icons/io/index``` and that solved it for me.

Basically, add "index" after the specific library you are using e.g ```react-icons/md``` should be ```react-icons/md/index```. That's all.
